wallet/txrules/rules.go

```diff
@@ -9,9 +9,31 @@ import (
 	"decred.org/dcrwallet/v2/errors"
 	"github.com/decred/dcrd/dcrutil/v4"
 	"github.com/decred/dcrd/txscript/v4"
+	"github.com/decred/dcrd/txscript/v4/stdscript"
 	"github.com/decred/dcrd/wire"
 )
 
+// StakeSubScriptType potentially transforms the provided script type by
+// converting the various stake-specific script types to their associated sub
+// type.  It will be returned unmodified otherwise.
+func StakeSubScriptType(scriptType stdscript.ScriptType) (stdscript.ScriptType, bool) {
```

Comment:
`txrules` is a somewhat odd place for this helper, which I essentially yanked out of an unexported version in dcrd, but `txrules` is a fairly ubiquitous import with relatively few of its own dcrd imports.  Alt. we can replicate the switch in the packages that need this logic.